Plugin/SalesRuleQuoteDiscountPlugin.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+namespace Bolt\Boltpay\Plugin;
+
+use Magento\Checkout\Model\Session as CheckoutSession;
+
+class SalesRuleQuoteDiscountPlugin
+{
+    /** @var CheckoutSession */
+    private $checkoutSession;
+    
+    public function __construct(
+        CheckoutSession $checkoutSession
+    ) {
+        $this->checkoutSession = $checkoutSession;
+    }
+    
+    public function beforeCollect(\Magento\SalesRule\Model\Quote\Discount $subject, $quote, $shippingAssignment, $total)
+    {
+        $this->checkoutSession->setBoltCollectSaleRuleDiscounts([]);
```

Comment:
@dumega When this actually happens?

If logged in user adds a coupon, logout and login again, will it work?

Reply:
@vitaliyreznikov good catch/. there is a bug when logout/login. I will look into it.

Reply:
@vitaliyreznikov After investigation, this is a bug with Amasty reward point (it is enabled on my dev site), and I have created a commit to fix it.
And for your concern, Mangeto would collect the address discount amount every time when refresh the cart page or via ajax, so this session data can be reset properly.